caddyhttp/httpserver/replacer.go

```diff
@@ -223,8 +223,22 @@ func (r *replacer) getSubstitution(key string) string {
 		}
 		return host
 	case "{path}":
-		return r.request.URL.Path
+		// if a rewrite has happened, the original URI should be used as the path
+		// rather than the rewritten URI
+		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
+		if path == "" {
+			path = r.request.URL.Path
+		}
+		return path
 	case "{path_escaped}":
+		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
```

Comment:
This Header field name is used in fastcgi.go, rewrite.go and now replacer.go . I tried creating it as an exported constant but ran into import cycle issues.

Should this be a const?  What file should the const be declared in?